.github/workflows/workspace-integration-tests.yml

```diff
@@ -0,0 +1,179 @@
+name: "Workspace integration tests"
+on:
+    workflow_dispatch:
```

Comment:
to test the workflow, you'd need to first have it exist on main. Once the workflow is on main, you can then trigger the workflow and run it from a branch (including the workflow changes on that branch). You could first merge an empty workflow with this name, and then iterate on a branch with your changes. But also sounds good to just merge this PR and then iterate on it